conn.go

```diff
@@ -204,43 +206,36 @@ func (c *Conn) readMessages(ctx context.Context) {
 
 		case m.response != nil:
 			resp := m.response
-			if resp != nil {
```

Comment:
The nested `if` is unnecessary, since the switch `case` just proved that the response is non-nil.